helpers.js

```diff
@@ -18,17 +23,54 @@ let getUserByEmail = function(email, users) {
 
 // filters the URLs for only that user
 let userURLs = function(user, urlDatabase) {
-  let userDatabase = { };
+  let userDatabase = {};
   for (let url in urlDatabase) {
     if (urlDatabase[url].userID === user) {
-      userDatabase[url] = {longURL: urlDatabase[url].longURL, userID: user};
+      userDatabase[url] = { longURL: urlDatabase[url].longURL, userID: user };
     }
   }
   return userDatabase;
 };
 
+const hashPassword = (password, saltRounds = 10) => {
+  let salt = bcrypt.genSaltSync(saltRounds);
+  return bcrypt.hashSync(password, salt);
+};
+
+const getEmailList = (stateUsers) => {
+  // returns an array with the VALUES (not KEYS) in stateUsers
+  const arrayOfUserObjects = Object.values(stateUsers);
+  // reduce is your best friend you can do pretty much everything with it
+  return arrayOfUserObjects.reduce(
+    (arrayToBeReturned, currentUserObject) => [...arrayToBeReturned, currentUserObject.email],
+    [],
+  );
+};
```

Comment:
Refactored to more declarative code by using array methods like reduce instead of looping with a `for`.
Also, using spreading syntax in line 45. The callback passed to reduce receives two arguments and returns an array of objects (users). This and other patterns with spreading syntax are used a lot in the wild these days so I'd definitely recommend you to practice and get familiar with their use.